library/ccm.c

```diff
@@ -121,30 +156,55 @@ void mbedtls_ccm_free( mbedtls_ccm_context *ctx )
 /*
  * Update the CBC-MAC state in y using a block in b
  * (Always using b as the source helps the compiler optimise a bit better.)
+ * Initial b masking happens outside of this macro due to various sources of it.
  */
 #define UPDATE_CBC_MAC                                                      \
     for( i = 0; i < 16; i++ )                                               \
-        y[i] ^= b[i];                                                       \
+    {                                                                       \
```

Comment:
I think we should add flow_control to loops and to mask generation. The same concept as I did in AES core implementation. But we can merge it as is and improve it later